src/renderer/src/routes/pages/Wallets/index.tsx

```diff
@@ -182,6 +229,69 @@ export const WalletsPage = () => {
           </footer>
         </section>
       )}
-    </PortfolioLayout>
+      <div className="flex-grow flex flex-col gap-y-5">
+        <div className="w-full flex bg-gray-800 rounded flex-grow drop-shadow-lg animate-pulse"></div>
+        <div className="w-full bg-gray-800 rounded flex-grow drop-shadow-lg overflow-scroll max-h-[20rem]">
+          <h1 className="text-white m-4">{commonActivity('title')}</h1>
+          <Separator className="mx-4" />
+          <InfiniteScroll
+            dataLength={transactions.length}
+            next={() => {}}
+            hasMore={true}
+            loader={<h4>Loading...</h4>}
+            endMessage={
+              <p style={{ textAlign: 'center' }}>
+                <b>Yay! You have seen it all</b>
+              </p>
+            }
+            refreshFunction={() => {}}
+          >
+            <table className="min-w-full text-xs text-left m-4 justify-evenly">
+              <thead>
+                <tr className="text-gray-100 opacity-75">
+                  <th className="pl-4 w-[5.5rem]">{commonActivity('date')}</th>
+                  <th className="w-[4.5rem]">{commonActivity('time')}</th>
+                  <th className="w-[5rem]">{commonActivity('name')}</th>
+                  <th className="w-[30.5rem]">{commonActivity('hash')}</th>
+                  <th></th>
+                </tr>
+              </thead>
+              <tbody>
+                {transactions.map((tr, index) => (
+                  <tr
+                    key={index}
+                    className={(() => {
+                      return StyleHelper.mergeStyles('', {
+                        'bg-gray-300 bg-opacity-15': index % 2 !== 0,
+                      })
+                    })()}
+                  >
+                    <td className="pl-4">{timeToDate(tr.time)}</td>
+                    <td>{timeToHour(tr.time)}</td>
+                    <td>{hashToContractName(tr.transfers[0].from, tr.transfers[0].contractHash)}</td>
+                    <td>
+                      <div className="flex items-center justify-between">
+                        {tr.hash}
+                        <IconButton
+                          icon={<MdContentCopy className="fill-neon w-6 h-6" />}
+                          onClick={() => handleCopyHash(tr.hash)}
+                          type="button"
+                        />
+                      </div>
+                    </td>
+                    <td>
+                      <div className="flex items-center">
+                        <span className="text-neon leading-[0.1rem]">{commonActivity('view')}</span>
+                        <TbChevronRight className="w-6 h-6 text-gray-300" />
+                      </div>
+                    </td>
+                  </tr>
+                ))}
+              </tbody>
+            </table>
+          </InfiniteScroll>
+        </div>
+      </div>
+    </MainLayout>
```

Comment:
What do you think about separating this into another component ? This would make this page easier to read and organized.

You don't need to create it inside the components folder, as it will not be reused, create it inside the folder on this page.

Reply:
This won't be the final one; adjustments are needed after the React Query above.